internal/xds/translator/translator.go

```diff
@@ -126,13 +128,18 @@ func (t *Translator) Translate(xdsIR *ir.Xds) (*types.ResourceVersionTable, erro
 	// Check if an extension want to inject any clusters/secrets
 	// If no extension exists (or it doesn't subscribe to this hook) then this is a quick no-op
 	if err := processExtensionPostTranslationHook(tCtx, t.ExtensionManager); err != nil {
-		errs = errors.Join(errs, err)
-		// Setting the configuration to fail open will mean that Envoy Gateway ignores the error and keeps the resources
-		// as they were before the extension server was called.
-		if t.ExtensionManager != nil && !(*t.ExtensionManager).FailOpen() {
+		// If the extension server returns an error, and the extension server is not configured to fail open,
+		// then replace all of the routes in the virtual host with a single route that returns an InternalServerError result.
+		// The original extension server error is suppressed.
+		if !(*t.ExtensionManager).FailOpen() {
```

Comment:
fail open becomes useless if we return an error to the xds-translator, since those changes will not push to the data plane, we should log here and not return an error imo

on a different note, we should rethink how to want to express error scenario tradeoffs and fail closed / fail open may not be able to sum it up properly